### ISSUE TITLE
logarithmic time performance

### ISSUE THREAD

**Issue by davidenitti** [BayesianLogic/blog#337]
Dear,
I am comparing Blog with my language, and I noticed a logarithmic behavior of the time performance of BLOG.
For example, if I use 1000 samples it takes 0.6 s, if I use 10000 samples it takes 1.5 s (instead of 0.6x10=6), if I use 100000 samples it takes 5.7 s (instead of 1.5x10=15). (excluding parsing time.)
I would aspect that the time is linear with respect to the number of samples. It also seems that there is no parallelization (initialization excluded). Am I right?
can you give an explanation of this?

**Comment by jxwuyi** [comment]
Hi,

What model are you using?

davidenitti <notifications@github.com>于2015年6月18日星期四写道：

**Comment by davidenitti** [comment]
id-uncert-det.blog
![id-uncert-det blog](https://cloud.githubusercontent.com/assets/7438184/8251414/a87559aa-167c-11e5-9801-5dd3fd573c96.png)
The time given the number of samples is logarithmic below 20000 samples then becomes linear. this is really strange.
this happens also for other models, e.g., burglary.blog
for DBLOG I noticed that there is multi-threading, for BLOG there is no multi-threading as far as I can see

**Comment by cberzan** [comment]
You're right that the runtime should be linear in the number of samples. I suspect the non-linear runtime you're seeing with small number of samples is because of data structure / JIT overhead. When you run with more samples, this overhead becomes less pronounced, and you see linear runtime.

We don't do explicit parallelism / multithreading in either BLOG or DBLOG. You may see multiple cores being used, but that's because of parallelism in the JVM, not in our code.
